### ISSUE TITLE
wrong image size on Ubuntu with Julia Display

### ISSUE THREAD

**Issue by piever** [barche/QML.jl#53]
Running "example/plot.jl" gives me this:

![qml](https://user-images.githubusercontent.com/6333339/35282825-6d64490c-004e-11e8-851c-cc7fa989d9d0.png)

I think it may be a linux specific problem (I installed QML using the libraries in #52).

**Comment by piever** [comment]
Update: it is actually not strictly a QML issue. The display size issue only happen with the GR backend. Gtk is also affected, for example:

```julia
using Plots, Gtk
gr()
plot(rand(100))
tmp = tempname() * ".png"
savefig(tmp)

img = GtkImage(tmp)

win = GtkWindow("test")
push!(win, img)
showall(win)
```
would show an image much bigger than the original png. It wouldn't happen with a different backend (have tried plotlyjs to confirm). I suspect it started with [#1318](https://github.com/JuliaPlots/Plots.jl/pull/1318), I'm pinging @jheinen as he probably knows what's going on.

**Comment by barche** [comment]
Hi,

Sorry for my late reply, but this is probably the device pixel ratio. The `gr.jl` example takes this into account. The trick is to pass the QML item to the callback, and then use the `effectiveDevicePixelRatio` function to get the ratio between image pixels and screen pixels.

**Comment by piever** [comment]
Hi, thanks for looking into this. I actually do not believe this has to do with `effectiveDevicePixelRatio` as it used to work fine on my Mac retina screen (which has an `effectiveDevicePixelRatio` of 2). I believe the issue is specific to GR due to the way  they save PNG, which changed recently (see for example [this comment](https://github.com/JuliaPlots/Plots.jl/issues/1355#issuecomment-357468044) ). Can you reproduce the problem if you check out the latest Plots and GR and try the example on your setup?

**Comment by jheinen** [comment]
There s no need to use PNGs in QML/GR environments. You can directly draw into the widget provided by Qt, which is much faster.

GR example: https://github.com/jheinen/GR.jl/blob/master/examples/qml_ex.qml
Plots + GR example: https://github.com/jheinen/GR.jl/blob/master/examples/qml_plots_ex.jl

**Comment by piever** [comment]
Thanks, I'll try that in my use case.

**Comment by jheinen** [comment]
See here: https://twitter.com/josef_heinen/status/812324559517052928

**Comment by piever** [comment]
Probably not the right place to report this, but I'm having issues running your example on Ubuntu:

```julia
julia> Pkg.status("GR")
 - GR                            0.25.0

julia> ENV["GRDIR"]=""
""

julia> Pkg.build("GR")

julia> GR.version()
"0.28.0"

julia> include(joinpath(Pkg.dir("GR", "examples"), "qml_plots_ex.jl"))
GKS: Qt5 support not compiled in
```

I get the QML window, with the slider, which detects my mouse position but no plot appears.

**Comment by jheinen** [comment]
We have to fix this in our CI environment or provide a method which would allow the user to build such extensions, e.g. the Qt5 plugin.

**Comment by piever** [comment]
Yes, I have an ubuntu machine (which has this issue) and a mac (on which I can run your QML+GR example). Thanks for looking into this!

**Comment by jheinen** [comment]
Are you using Ubuntu, Hombrew or the official qt.io Qt5 ?

**Comment by piever** [comment]
I installed QML libraries on my ubuntu machine using this command:

 `sudo apt-get -y install cmake cmake-data qtdeclarative5-dev qml-module-qtquick-controls qml-module-qtquick-dialogs qml-module-qt-labs-folderlistmodel qml-module-qt-labs-settings qmlscene qt5-default`

**Comment by piever** [comment]
Update: I've noticed that, at least on Gtk, the GR plot saved as svg displays correctly as a GtkImage. I was  wondering whether it's possible to change the `JuliaDisplay` type in QML to use svg format rather than png to display the plot. In Plots.jl language that would be `mimeformat` "image/svg+xml"  rather than "image/png". If you believe that for some backends png is better (which may well be) there could be an attribute of `JuliaDisplay` that allows the user to choose between `svg` and `png`.

**Comment by jheinen** [comment]
In the meantime, you can use native GR/Qt graphics with QML, as shown in [this](https://github.com/jheinen/GR.jl/blob/master/examples/qml_ex.jl) GR example.
![qml gr_on_ubuntu](https://user-images.githubusercontent.com/5718244/41591725-d97a4436-73ba-11e8-8981-6bbe7b401d23.png)

So, there is no need for PNGs or SVGs to display graphics in QML. The above method is much faster!
